sdk/core/Azure.Core/tests/HttpPipelineBuilderTest.cs

```diff
@@ -77,6 +78,66 @@ public async Task VersionDoesntHaveCommitHash()
                                  Regex.Escape($" ({RuntimeInformation.FrameworkDescription}; {RuntimeInformation.OSDescription})"), userAgent);
         }
 
+        [Test]
+        public async Task CustomClientRequestIdAvailableInPerCallPolicies()
+        {
+            var policy = new Mock<HttpPipelineSynchronousPolicy>();
+            policy.CallBase = true;
+            policy.Setup(p => p.OnSendingRequest(It.IsAny<HttpMessage>()))
+                .Callback<HttpMessage>(message =>
+                {
+                    Assert.AreEqual("ExternalClientId",message.Request.ClientRequestId);
+                    Assert.True(message.Request.TryGetHeader("x-ms-client-request-id", out string requestId));
+                    Assert.AreEqual("ExternalClientId", requestId);
+                }).Verifiable();
+
+            var options = new TestOptions();
+            options.Transport = new MockTransport(new MockResponse(200));
+            options.AddPolicy(policy.Object, HttpPipelinePosition.PerCall);
+
+            var pipeline = HttpPipelineBuilder.Build(options);
+            using (Request request = pipeline.CreateRequest())
+            {
+                request.Method = RequestMethod.Get;
+                request.Uri.Reset(new Uri("http://example.com"));
+                request.Headers.Add("x-ms-client-request-id", "ExternalClientId");
+                await pipeline.SendRequestAsync(request, CancellationToken.None);
+            }
+
+            policy.Verify();
+        }
+
+        [Test]
+        public async Task CustomClientRequestIdSetInPerCallPolicyAppliedAsAHeader()
```

Comment:
Sorry for causing this problem - but thank you for making the scenario work.

Reply:
It's a good test I with I had before.